File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    public int search(int[] nums, int target) {
        int minIndex=0, maxIndex=nums.length-1;
        while (minIndex<=maxIndex) {
            int midIndex = (minIndex + maxIndex) / 2;
            if(nums[midIndex] == target) {
                return midIndex;
            } else if (nums[midIndex] < target) {
                minIndex = midIndex+1;
            } else {
                maxIndex = midIndex-1;
            }
        }
        return -1;
    }
}

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        d = {}
        def dp(i):
            if i >= len(nums) or i < 0:
                return 0
            if i in d:
                return d[i]
            d[i] = max(nums[i] + dp(i+2), dp(i+1))
            return d[i]

        return dp(0)